cl/lib/elasticsearch_utils.py

```diff
@@ -1789,6 +1789,76 @@ def merge_unavailable_fields_on_parent_document(
                         result["id"], ""
                     )
 
+        case (
+            SEARCH_TYPES.RECAP | SEARCH_TYPES.DOCKETS
+        ) if request_type == "frontend":
+            # Merge initial complaint button to the frontend search results.
+            docket_ids = {doc["docket_id"] for doc in results}
+            # This query retrieves initial complaint documents considering two
+            # possibilities:
+            # 1. For district, bankruptcy, and appellate entries where we don't know
+            #    if the entry contains attachments, it considers:
+            #    document_number=1 and attachment_number=None and document_type=PACER_DOCUMENT
+            #    This represents the main document with document_number 1.
+            # 2. For appellate entries where the attachment page has already been
+            #    merged, it considers:
+            #    document_number=1 and attachment_number=1 and document_type=ATTACHMENT
+            #    This represents document_number 1 that has been converted to an attachment.
+            initial_complaints = (
+                RECAPDocument.objects.filter(
+                    QObject(
+                        QObject(
+                            attachment_number=None,
+                            document_type=RECAPDocument.PACER_DOCUMENT,
+                        )
+                        | QObject(
+                            attachment_number=1,
+                            document_type=RECAPDocument.ATTACHMENT,
+                        )
```

Comment:
Would this pull back attachment 1 for entry 1 in district courts too?

Reply:
Oh, yeah, this was retrieving `document:1 attachment:1` in district and bankruptcy courts as the initial complaint when it was available. I’ve fixed it so that this condition only applies to appellate RECAPDocuments.

This approach still requires only one additional query per page.

Thanks.